### ISSUE TITLE
I am new to TestNG framework, I'm doing some programs, I installed the TestNG version 6.14.3 and now studying TestNG framework, while starting the executing the test case are running properly and executing, but I am getting error (java.util.ConcurrentModificationException) 

### ISSUE THREAD

**Issue by Thiyagarajan9977** [trivago/cucable-plugin#175]
**Describe the bug**
I installed the TestNG version 6.14.3 and now studying TestNG framework, while starting the executing the test case are running properly and executing, but I am getting error (java.util.ConcurrentModificationException) 

![03-19-2021](https://user-images.githubusercontent.com/77095923/111766508-a00edf80-88cb-11eb-8fe0-c9366ce88b4b.png)

**To Reproduce**
1. Create a scenario with...
2. Run Cucable
3. See error

**Expected behavior**
My program running properly, But always getting this error (java.util.ConcurrentModificationException)

**Attachments**
If applicable, add an attachment to help explain your problem.

**Additional context**

**Comment by laxersaz** [comment]
If your scenarios run in parallel and you try to report them at the same time this probably happens. But this is not a Cucable problem but a problem of your test framework which is out of scope here.
